src/components/title-list-item/title-list-item.test.js

```diff
@@ -0,0 +1,181 @@
+    contributors: [
+      {
+        type: 'test-contributor-type',
+        contributor: 'test-contributor-name',
+      }
+    ],
+    identifiers: [
+      {
+        id: 'test-id',
+        subtype: 'test-subtype',
+        type: 'test-type',
+      }
+    ],
+    isSelected: false,
+    name: 'test-name',
+    tags: {
+      tagList: testTags,
+    },
+    visibilityData: {
+      isHidden: false,
+    },
+  },
+};
+
+describe('Given TitleListItem', () => {
+  const mockOnClick = jest.fn();
+
+  const renderTitleListItem = ({ ...props }) => render(
+    <Router history={routerHistory}>
+      <TitleListItem {...props} />
+    </Router>
+  );
+
+  it('should render skeleton when item prop is not provided', () => {
+    const { getByTestId } = renderTitleListItem(testPropsWithoutItem);
+
+    expect(getByTestId('skeleton-element')).toBeDefined();
+  });
+
+  it('should render TitleListItem link', () => {
+    const { getByTestId } = renderTitleListItem(testBasicProps);
+
+    expect(getByTestId('title-list-item-link')).toBeDefined();
+  });
+
+  it('should render TitleListItem headline', () => {
+    const { getByText } = renderTitleListItem(testBasicProps);
+
+    expect(getByText('test-name')).toBeDefined();
+  });
+
+  it('should not invoke onClick callback', () => {
+    renderTitleListItem(testBasicProps);
+    fireEvent.click(screen.getByRole('link', { name: /test-name/ }));
+
+    expect(mockOnClick).not.toBeCalled();
+  });
+
+  it('should invoke onClick callback', () => {
+    const testPropsWithOnClick = {
+      ...testBasicProps,
+      onClick: mockOnClick,
+    };
+
+    renderTitleListItem(testPropsWithOnClick);
+
+    fireEvent.click(screen.getByRole('link', { name: /test-name/ }));
+
+    expect(mockOnClick).toBeCalled();
+  });
+
+  it('should display publication type and publisher name', () => {
+    const testExtendedProps = {
+      ...testBasicProps,
+      item: {
+        ...testBasicProps.item,
+        publicationType: 'test-publication-type',
+        publisherName: 'test-publisher-name',
+      },
+      showPublisherAndType: true,
+      showContributors: true,
+    };
+
+    const { getByText } = renderTitleListItem(testExtendedProps);
+
+    expect(getByText('test-publication-type')).toBeDefined();
+    expect(getByText('test-publisher-name')).toBeDefined();
+  });
+
+  it('should display contributors', () => {
+    const testExtendedProps = {
+      ...testBasicProps,
+      showPublisherAndType: true,
+      showContributors: true,
+    };
+
+    const { getByText } = renderTitleListItem(testExtendedProps);
+
+    expect(getByText('test-contributor-name')).toBeDefined();
+  });
+
+  it('should display identifiers', () => {
+    const testExtendedProps = {
+      ...testBasicProps,
+      showIdentifiers: true,
+    };
+
+    const { getByText } = renderTitleListItem(testExtendedProps);
+
+    expect(getByText('IdentifiersList component')).toBeDefined();
+  });
+
+  describe('item metadata class', () => {
```

Comment:
I'm not sure that you need this `describe`. Actually, describe should starts with word `when` and logically separate tests. I've recomended to add all props that you add to `testExtendedProps` in test set as default. In this case you don't need to set this data each time when you write test.